core/templates/domain/summary/exploration-summary-backend-api.service.ts

```diff
@@ -17,77 +17,107 @@
+    if (!explorationIds.every(expId =>
+      this.validatorsService.isValidExplorationId(expId, true))) {
+      this.alertsService.addWarning('Please enter a valid exploration ID.');
+      let returnValue = [];
+      for (let i = 0; i < explorationIds.length; i++) {
+        returnValue.push(null);
       }
-
-      var explorationSummaryDataUrl = EXPLORATION_SUMMARY_DATA_URL_TEMPLATE;
-
-      $http.get(explorationSummaryDataUrl, {
-        params: {
-          stringified_exp_ids: JSON.stringify(explorationIds),
-          include_private_explorations: JSON.stringify(
-            includePrivateExplorations)
-        }
-      }).then(function(response) {
-        var summaries = angular.copy(response.data.summaries);
+      if (errorCallback) {
+        errorCallback(returnValue);
+      }
+      return;
+    }
+    const explorationSummaryDataUrl =
+    AppConstants.EXPLORATION_SUMMARY_DATA_URL_TEMPLATE;
+    this.httpClient.get(explorationSummaryDataUrl, {
+      params: {
+        stringified_exp_ids: JSON.stringify(explorationIds),
+        include_private_explorations: JSON.stringify(
+          includePrivateExplorations)
+      }
+    }).toPromise().then((summaries: ExplorationSummaryBackendDict) => {
+      try {
         if (successCallback) {
           if (summaries === null) {
-            var summariesError = (
+            const summariesError = (
               'Summaries fetched are null for explorationIds: ' + explorationIds
             );
             throw new Error(summariesError);
+          } else if (typeof (summaries) === 'string') {
+            const fetchingError = (
+              'Error on loading public exploration summaries.');
+            throw new Error(fetchingError);
           }
           successCallback(summaries);
         }
-      })['catch'](function(errorResponse) {
+      } catch (errorResponse) {
         if (errorCallback) {
-          errorCallback(errorResponse.data || errorResponse);
+          errorCallback(errorResponse);
         }
-      });
-    };
-
-    return {
-      /**
-       * Fetches a list of public exploration summaries and private
-       * exploration summaries for which the current user has access from the
-       * backend for each exploration ID provided. The provided list of
-       * exploration summaries are in the same order as input exploration IDs
-       * list, though some may be missing (if the exploration doesn't exist or
-       * or the user does not have access to it).
-       */
-      loadPublicAndPrivateExplorationSummaries: function(explorationIds) {
-        return $q(function(resolve, reject) {
-          _fetchExpSummaries(explorationIds, true, resolve, reject);
-        });
-      },
-      loadPublicExplorationSummaries: function(explorationIds) {
-        return $q(function(resolve, reject) {
-          _fetchExpSummaries(explorationIds, false, resolve, reject);
-        });
       }
-    };
+    }, (errorResponse) =>{
+      if (errorCallback) {
```

Comment:
I don't think this if condition is required.

Reply:
Removed!